temporal-sdk/src/main/java/io/temporal/internal/worker/LocalActivityWorker.java

```diff
@@ -232,6 +230,14 @@ public Throwable wrapFailure(Task task, Throwable failure) {
       int attempt = activityTask.getAttempt();
       result.setAttempt(attempt);
 
+      // Do not retry for non-retryable ApplicationFailures
+      if (result.getTaskFailed() != null
```

Comment:
Let's put this shebang
```
result.getTaskFailed() != null
          && result.getTaskFailed().getFailure() != null
          && result.getTaskFailed().getFailure() instanceof ApplicationFailure
          && ((ApplicationFailure) result.getTaskFailed().getFailure()).isNonRetryable()
```
into a separate function. This will make code self-documenting.